MetaMask/metamask-extension
# Update to component-library readme

## Explanation
Currently our `component-library` readme is a bit out of date. This PR updates with latest architecture, figma links and typescript initiatives 


## Screenshots/Screencaps

### Before
<img width="1440" alt="Screenshot 2023-04-06 at 5 27 29 PM" src="https://user-images.githubusercontent.com/8112138/230516714-1bab314f-e2ab-45d0-8629-e42ef2d92e95.png">

### After

https://user-images.githubusercontent.com/8112138/230517244-fcfc482c-632a-483d-a3aa-7b9205461f5f.mov

## Manual Testing Steps

- Go to the latest build of storybook on this PR
- Search `Introduction`
- Select the second option in component library
- Read to make sure it makes sense and check links

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
